resources/tweets_dump.py

```diff
@@ -0,0 +1,104 @@
+import csv
+import json
+import math
+import sys
+from time import sleep
+
+import tweepy
+
+'''
+    @license
+
+    Copyright(c) 2018, GarridoLabs and the project's contributors.
+
+    This source code is licensed under the Apache License, Version 2.0 found in
+    the LICENSE.txt file in the root directory of this source tree.
+'''
+
+
+class TweetsDump:
+    def __init__(self, credentials_path):
+        with open(credentials_path, "r") as data_file:
+            data_loaded = json.load(data_file)
+
+        # Twitter API credentials
+        self.consumer_key = data_loaded['apiKey']
+        self.consumer_secret = data_loaded['apiSecretKey']
+        self.access_key = data_loaded['accessToken']
+        self.access_secret = data_loaded['accessTokenSecret']
+
+        auth = tweepy.OAuthHandler(self.consumer_key, self.consumer_secret)
+        auth.set_access_token(self.access_key, self.access_secret)
+        self.api = tweepy.API(auth)
+
+    @staticmethod
+    def is_retweet(tweet):
+        return tweet["retweeted"]
+
+    @staticmethod
+    def get_ids(ids_input_file):
+        with open(ids_input_file) as f:
+            ids = json.load(f)
+
+        print('total ids: {}'.format(len(ids)))
+
+        return ids
+
+    @staticmethod
+    def write_non_retweeted_tweets_plain_text(output_file, tweets):
+        f = csv.writer(
```

Comment:
Write a file using this kind of methods is not often safe. I think it would be better as:

```
with open(output_file, 'w', newline='\n') as csvfile:
    f = csv.writer(csvfile, quoting=csv.QUOTE_NONE)
    for tweet in tweets:
        if not TweetsDump.is_retweet(tweet):
            f.writerow([tweet["text"]])
```

Reply:
Thank you! I already change it